### PR TITLE
test/authService implement test

### PR DESCRIPTION
**Pull request by paul2126**
#### PR Description:
- Implement authService test

##### Changes Included:
- [x] Added new feature(s)
- [ ] Fixed identified bug(s)
- [ ] Updated relevant documentation

##### Notes for Reviewer:
Any specific instructions or points to be considered by the
reviewer.

#### Reviewer Checklist:
- [x] Code is written in clean, maintainable, and idiomatic form.
- [x] Automated test coverage is adequate.
- [x] All existing tests pass.
- [x] Manual testing has been performed to ensure the PR works as expected.
- [x] Code review comments have been addressed or clarified.

#### Additional Comments:
Add any other comments or information that might be useful for the
review process.

### CODE REVIEW

File: frontend/app/src/test/java/com/example/speechbuddy/service/AuthServiceTest.kt [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+import retrofit2.Response
+
+class AuthServiceTest {
+    private lateinit var authService: AuthService
+    private val mockEmail = "test@example.com"
+    private val mockPassword = "password123"
+    private val mockNickname = "TestUser"
+    private val mockCode = "123456"
+    private val errorResponseBody =
+        "{\"error\":\"Something went wrong\"}".toResponseBody("application/json".toMediaType())
```
Comment (sukchan-0811) [inline frontend/app/src/test/java/com/example/speechbuddy/service/AuthServiceTest.kt:28]:
```suggestion
class AuthServiceTest {

    private lateinit var authService: AuthService
    
    private val mockEmail = "test@example.com"
    private val mockPassword = "password123"
    private val mockNickname = "TestUser"
    private val mockCode = "123456"
    
    private val errorResponseBody =
        "{\"error\":\"Something went wrong\"}".toResponseBody("application/json".toMediaType())
```

File: frontend/app/src/test/java/com/example/speechbuddy/service/AuthServiceTest.kt [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+
+    @Before
+    fun setUp() {
+        authService = mockk()
+    }
+
```
Comment (sukchan-0811) [inline frontend/app/src/test/java/com/example/speechbuddy/service/AuthServiceTest.kt:34]: 여기는 test 끝날 때 mock한 거 cleanup하는 작업은 필요없나요?

File: frontend/app/src/test/java/com/example/speechbuddy/service/AuthServiceTest.kt [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+        val response = authService.signup(signupRequest)
+
+        coVerify(exactly = 1) { authService.signup(signupRequest) }
+        assertFalse(response.isSuccessful)
+    }
+
+    @Test
+    fun `should return response with error when login request is invalid`() = runBlocking {
+        val loginRequest = AuthLoginRequest(mockEmail, "wrongpassword")
```
Comment (sukchan-0811) [inline frontend/app/src/test/java/com/example/speechbuddy/service/AuthServiceTest.kt:139]:
```suggestion
        val loginRequest = AuthLoginRequest(mockEmail, "wrong_password")
```

File: frontend/app/src/test/java/com/example/speechbuddy/service/AuthServiceTest.kt [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+        coEvery { authService.login(loginRequest) } returns Response.error(401, errorResponseBody)
+
+        val response = authService.login(loginRequest)
+
+        coVerify(exactly = 1) { authService.login(loginRequest) }
+        assertFalse(response.isSuccessful)
+    }
+
+    @Test
+    fun `should return response with error when reques temail send is invalid for signup`() = runBlocking {
```
Comment (sukchan-0811) [inline frontend/app/src/test/java/com/example/speechbuddy/service/AuthServiceTest.kt:149]:
```suggestion
    fun `should return response with error when request email send is invalid for signup`() = runBlocking {
```

File: frontend/app/src/test/java/com/example/speechbuddy/service/AuthServiceTest.kt [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+
+        val response = authService.verifySendPW(verifyEmailSendRequest)
+
+        coVerify(exactly = 1) { authService.verifySendPW(verifyEmailSendRequest) }
+        assertFalse(response.isSuccessful)
+    }
+
+    @Test
+    fun `should return response with error when request code is invalid for signup`() = runBlocking {
+        val verifyEmailAcceptRequest = AuthVerifyEmailAcceptRequest(mockEmail, "wrongcode")
```
Comment (sukchan-0811) [inline frontend/app/src/test/java/com/example/speechbuddy/service/AuthServiceTest.kt:178]:
```suggestion
        val verifyEmailAcceptRequest = AuthVerifyEmailAcceptRequest(mockEmail, "wrong_code")
```

File: frontend/app/src/test/java/com/example/speechbuddy/service/AuthServiceTest.kt [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+
+        val response = authService.verifyAcceptSignup(verifyEmailAcceptRequest)
+
+        coVerify(exactly = 1) { authService.verifyAcceptSignup(verifyEmailAcceptRequest) }
+        assertFalse(response.isSuccessful)
+    }
+
+    @Test
+    fun `should return response with error when request code is invalid for password reset`() = runBlocking {
+        val verifyEmailAcceptRequest = AuthVerifyEmailAcceptRequest(mockEmail, "wrongcode")
```
Comment (sukchan-0811) [inline frontend/app/src/test/java/com/example/speechbuddy/service/AuthServiceTest.kt:192]:
```suggestion
        val verifyEmailAcceptRequest = AuthVerifyEmailAcceptRequest(mockEmail, "wrong_code")
```

File: frontend/app/src/test/java/com/example/speechbuddy/service/AuthServiceTest.kt [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+        coEvery { authService.resetPassword(resetPasswordRequest) } returns Response.error(
+            400,
+            errorResponseBody
+        )
+
+        val response = authService.resetPassword(resetPasswordRequest)
+
+        coVerify(exactly = 1) { authService.resetPassword(resetPasswordRequest) }
+        assertFalse(response.isSuccessful)
+    }
```
Comment (sukchan-0811) [inline frontend/app/src/test/java/com/example/speechbuddy/service/AuthServiceTest.kt:216]:
```suggestion
    }
    
```

**Review by sukchan-0811 — COMMENTED** [review]
코드 구조 좀 바꾸고 warning 안 뜨게 오타 같은 거 고쳤습니다

중요: @After 처리 작업 필요 없는지 확인 한번 부탁드립니다!

File: frontend/app/src/test/java/com/example/speechbuddy/service/AuthServiceTest.kt (hunk 2 above)
Comment (paul2126) [reply frontend/app/src/test/java/com/example/speechbuddy/service/AuthServiceTest.kt:37]: 네 기본적으로 inline mock 된 것들은 JVM이 알아서 처리한다고 하네요. 그리고 dispatcher를 쓰거나 아니면 network connection, database connection같은 것들을 닫는 위주로 쓰인다고 해서 여기서는 `@after`가 필요 없을 것 같아요

**Review by paul2126 — COMMENTED** [review]
사소한 부분까지 확인해주셔서 감사해요!

**Pull request merged**